src/include/cephfs/ceph_ll_client.h

```diff
@@ -110,28 +110,77 @@ struct ceph_statx {
 #define FALLOC_FL_PUNCH_HOLE 0x02
 #endif
 
-/* delegation recalls */
+/** ceph_deleg_cb_t: Delegation recalls
+ *
+ * Called when there is an outstanding Delegation and there is conflicting
+ * access, either locally or via cap activity.
+ * @fh: open filehandle
+ * @priv: private info registered when delegation was acquired
+ */
 typedef void (*ceph_deleg_cb_t)(Fh *fh, void *priv);
 
-/* inode data/metadata invalidation */
+/**
+ * client_ino_callback_t: Inode data/metadata invalidation
+ *
+ * Called when the client wants to invalidate the cached data for a range
+ * in the file.
+ * @handle: client callback handle
+ * @ino: vino of inode to be invalidated
+ * @off: starting offset of content to be invalidated
+ * @len: length of region to invalidate
+ */
 typedef void (*client_ino_callback_t)(void *handle, vinodeno_t ino,
 	      int64_t off, int64_t len);
 
-/* dentry invalidation */
+/**
+ * client_dentry_callback_t: Dentry invalidation
+ *
+ * Called when the client wants to purge a dentry from its cache.
+ * @handle: client callback handle
+ * @dirino: vino of directory that contains dentry to be invalidate
+ * @ino: vino of inode attached to dentry to be invalidated
+ * @name: name of dentry to be invalidated
+ * @len: length of @name
+ */
 typedef void (*client_dentry_callback_t)(void *handle, vinodeno_t dirino,
 					 vinodeno_t ino, const char *name,
 					 size_t len);
 
-/* remount entire fs */
+/**
+ * client_remount_callback_t: Remount entire fs
+ *
+ * Called when the client needs to purge the dentry cache and the application
+ * doesn't have a way to purge an individual dentry. Mostly used for ceph-fuse
+ * on older kernels.
+ * @handle: client callback handle
+ */
+
 typedef int (*client_remount_callback_t)(void *handle);
 
-/* lock request interrupted */
+/**
+ * client_switch_interrupt_callback_t: Lock request interrupted
+ *
+ * Called before and after sleeping while waiting for a file lock request.
+ * @handle: client callback handle
+ * @data: opaque data passed to interrupt before
```

Comment:
Grammar nit: Was there something to come after the word "before"? If not, it is phrased a bit awkwardly.

Reply:
Maybe be the "both before and after...." to be more clear.

Reply:
Hopefully clearer now!

Reply:
Indeed it is, thanks!